### ISSUE TITLE
How can I set ConfigurationMetrics using RegistryManager

### ISSUE THREAD

**Issue by ggthename** [Azure/azure-iot-sdk-java#600]
- **OS and version used:** <VERSION> 
maxOS
- **Java runtime used:** <VERSION> 
java 8
- **SDK version used:** <VERSION> 

# Description of the issue:

I want to apply metrics queries for IoT Hub configuration using RegistryManager

# Code sample exhibiting the issue:

```
RegistryManager registryManager = RegistryManager.createFromConnectionString(serviceDto.getAzureIotHub().getAzureIotHubConnectionString());
        Configuration configuration = new Configuration(azureDeviceFirmwareUpdateInfo.getConfigurationId());

        ConfigurationContent configurationContent = new ConfigurationContent();
        configurationContent.setDeviceContent(azureDeviceFirmwareUpdateInfo.firmwareUpdateInfoToMap());

        ConfigurationMetrics configurationMetrics = new ConfigurationMetrics();
        configurationMetrics.setQueries( azureDeviceFirmwareUpdateInfo.getMetricQueries() );

        configuration.setContent(configurationContent);
        // How can I set configurationMetircs ?
        configuration.setTargetCondition("tags.serviceCode= \'"+serviceCode+"\'"+
                                        " AND " +
                                        " tags.deviceModelCode = \'"+deviceModelCode+"\'");
        configuration.setPriority(20);

        registryManager.addConfiguration(configuration);
```

If I use javascript SDK, I can handle it like this.

```
// <configuration>
var firmwareConfig = {
  id: sampleConfigId,
  content: {
    deviceContent: {
      'properties.desired.firmware': {
        fwVersion: fwVersion,
        fwPackageURI: fwPackageURI,
        fwPackageCheckValue: fwPackageCheckValue
      }
    }
  },

  // Maximum of 5 metrics per configuration
  metrics: {
    queries: {
      current: 'SELECT deviceId FROM devices WHERE configurations.[['+sampleConfigId+']].status=\'Applied\' AND properties.reported.firmware.fwUpdateStatus=\'current\'',
      applying: 'SELECT deviceId FROM devices WHERE configurations.[['+sampleConfigId+']].status=\'Applied\' AND ( properties.reported.firmware.fwUpdateStatus=\'downloading\' OR properties.reported.firmware.fwUpdateStatus=\'verifying\' OR properties.reported.firmware.fwUpdateStatus=\'applying\')',
      rebooting: 'SELECT deviceId FROM devices WHERE configurations.[['+sampleConfigId+']].status=\'Applied\' AND properties.reported.firmware.fwUpdateStatus=\'rebooting\'',
      error: 'SELECT deviceId FROM devices WHERE configurations.[['+sampleConfigId+']].status=\'Applied\' AND properties.reported.firmware.fwUpdateStatus=\'error\'',
      rolledback: 'SELECT deviceId FROM devices WHERE configurations.[['+sampleConfigId+']].status=\'Applied\' AND properties.reported.firmware.fwUpdateStatus=\'rolledback\''
    }
  },

  // Specify the devices the firmware update applies to
  targetCondition: 'tags.devicetype = \'chiller\'',
  priority: 20
};
// </configuration>
```

But I couldn't find Java SDK for applying ConfigurationMetrics.
Can you help me?
Thank you.

# Console log of the issue:

# Need Support?
* Have a feature request for SDKs? Please post it on [User Voice](https://feedback.azure.com/forums/321918-azure-iot) to help us prioritize.
* Have a technical question? Ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/azure-iot-hub) with tag “azure-iot-hub”
* Need Support? Every customer with an active Azure subscription has access to support with guaranteed response time.  Consider submitting a ticket and get assistance from Microsoft support team
* Found a bug? Please help us fix it by thoroughly documenting it and filing an issue on GitHub (C, Java, .NET, Node.js, Python).

**Comment by timtay-microsoft** [comment]
Right now, it looks like we don't allow you to set the ConfigurationMetrics within your Configuration object. That doesn't seem right to me, so I'll work to extend a setter API on the Configuration object that allows you to set the ConfiguraitonMetrics

**Comment by ggthename** [comment]
Thank you! I'm waiting for your release.

**Comment by timtay-microsoft** [comment]
Sorry for the delay on the release. This feature was just released in Service Client 1.19.0, so I'm going to close this issue now. Thanks for bringing this issue to our attention!

**Comment by az-iot-builder-01** [comment]
@ggthename, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
